### PR TITLE
Feature/7 training screen

### PR DESCRIPTION
**Pull request by BrayanRGarciaM**

### CODE REVIEW

File: .flutter-plugins-dependencies [hunk 1]
```diff
@@ -1 +1 @@
...
\ No newline at end of file
+{"info":"This is a generated file; do not edit or check into version control.","plugins":{"ios":[{"name":"flutter_blue_plus","path":"C:\\\\Users\\\\ASUS\\\\Desktop\\\\LaZona\\\\flutter\\\\.pub-cache\\\\hosted\\\\pub.dartlang.org\\\\flutter_blue_plus-1.2.0\\\\","native_build":true,"dependencies":[]},{"name":"hexcolor","path":"C:\\\\Users\\\\ASUS\\\\Desktop\\\\LaZona\\\\flutter\\\\.pub-cache\\\\hosted\\\\pub.dartlang.org\\\\hexcolor-2.0.7\\\\","native_build":true,"dependencies":[]}],"android":[{"name":"flutter_blue_plus","path":"C:\\\\Users\\\\ASUS\\\\Desktop\\\\LaZona\\\\flutter\\\\.pub-cache\\\\hosted\\\\pub.dartlang.org\\\\flutter_blue_plus-1.2.0\\\\","native_build":true,"dependencies":[]},{"name":"hexcolor","path":"C:\\\\Users\\\\ASUS\\\\Desktop\\\\LaZona\\\\flutter\\\\.pub-cache\\\\hosted\\\\pub.dartlang.org\\\\hexcolor-2.0.7\\\\","native_build":true,"dependencies":[]}],"macos":[],"linux":[],"windows":[],"web":[]},"dependencyGraph":[{"name":"flutter_blue_plus","dependencies":[]},{"name":"hexcolor","dependencies":[]}],"date_created":"2022-08-09 19:50:50.442048","version":"3.0.2"}
```
Comment (juliomejia97) [inline .flutter-plugins-dependencies:1]: Please, review this file because is not necessary to be added

File: lib/main.dart [hunk 2]
```diff
@@ -9,11 +10,25 @@ void main() async {
...
+  static const greenColor = Color(0xff3cab1f);
+  static const purpleColor = Color(0xFF753A88);
+}
+
+class ZoneDimens {
+  static const appbarMainSize = 50.0;
+  static const upperTextSize = 70.0;
+  static const mediumTextSize = 20.0;
+  static const minTextSize = 10.0;
 }
```
Comment (juliomejia97) [inline lib/main.dart:32]: Create a separate file in core.

File: lib/src/features/devices/presentation/training/statistic/pass/overall_pass.dart [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return Column(children: [
+      Text(widget.categoryValue,
+          style: const TextStyle(
+              color: ZoneTheme.fontColor, fontSize: ZoneDimens.minTextSize)),
+      Text(widget.titleCategory,
+          style: const TextStyle(
+              color: ZoneTheme.fontColor, fontSize: ZoneDimens.minTextSize))
+    ]);
```
Comment (juliomejia97) [inline lib/src/features/devices/presentation/training/statistic/pass/overall_pass.dart:29]: Please add format and make more bigger the text

File: lib/src/features/devices/presentation/training/statistic/pass/overall_pass.dart [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+class OverallPassStatisticState extends State<OverallPassStatistic> {
+  @override
+  Widget build(BuildContext context) {
+    return Column(children: [
+      Text(widget.categoryValue,
+          style: const TextStyle(
+              color: ZoneTheme.fontColor, fontSize: ZoneDimens.minTextSize)),
+      Text(widget.titleCategory,
```
Comment (juliomejia97) [inline lib/src/features/devices/presentation/training/statistic/pass/overall_pass.dart:26]:
```suggestion
     const Text(widget.titleCategory,
```

File: lib/src/features/devices/presentation/training/statistic/precision/wall_statistic.dart [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+class WallStatisticState extends State<WallStatistic> {
+  @override
+  Widget build(BuildContext context) {
+    return Flexible(
+      flex: 3,
+      fit: FlexFit.loose,
+      child: SizedBox(
+        height: MediaQuery.of(context).size.height * 0.1,
```
Comment (juliomejia97) [inline lib/src/features/devices/presentation/training/statistic/precision/wall_statistic.dart:26]: Please, make it a little beat less high
```suggestion
        height: MediaQuery.of(context).size.height * 0.09,
```

File: lib/src/features/devices/presentation/training/statistic/precision/precision.dart [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  State<StatefulWidget> createState() {
+    return PrecisionStatisticState();
+  }
+}
+
+class PrecisionStatisticState extends State<PrecisionStatistic> {
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      margin: const EdgeInsets.only(left: 20, right: 10, top: 10),
```
Comment (juliomejia97) [inline lib/src/features/devices/presentation/training/statistic/precision/precision.dart:21]: Please, make it more center the wall. Maybe reduce the left.

File: lib/src/features/devices/presentation/training/statistic/statistics.dart [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+        mainAxisSpacing: 8.0,
+        children: [
+          StatisticCard(NUMBER_OF_PASSES, Icons.e_mobiledata,
+              children: PassStatistic()),
+          StatisticCard(TIME_PER_TOUCHES, Icons.e_mobiledata,
+              children: TimePerTouch()),
+          StatisticCard(PRECISION, Icons.e_mobiledata,
+              children: PrecisionStatistic()),
+          StatisticCard(FORCE_CONSISTENCY, Icons.e_mobiledata,
+              children: ForceConsistency())
```
Comment (juliomejia97) [inline lib/src/features/devices/presentation/training/statistic/statistics.dart:38]: 🤩

File: lib/src/features/devices/presentation/training/statistic/statistics.dart [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+        ],
+      ),
+    );
+  }
+}
+
+//TODO: Move to an other widget
+class StatisticCard extends StatelessWidget {
```
Comment (juliomejia97) [inline lib/src/features/devices/presentation/training/statistic/statistics.dart:46]: Please, move this to another widget

File: lib/src/features/devices/presentation/training/statistic/time_per_touch/time_per_touch.dart [hunk 9]
```diff
@@ -0,0 +1,85 @@
+import 'package:flutter/material.dart';
+import 'package:syncfusion_flutter_gauges/gauges.dart';
+import 'package:zone_app/main.dart';
+import 'package:zone_app/src/core/strings.dart';
+
+import '../statistics.dart';
+
+class TimePerTouch extends StatefulWidget {
```
Comment (juliomejia97) [inline lib/src/features/devices/presentation/training/statistic/time_per_touch/time_per_touch.dart:8]: 👍 Brutal!

File: lib/src/features/devices/presentation/training/training_screen.dart [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+          child: Icon(
+            Icons.mode_standby_rounded,
+            color: ZoneTheme.whiteColor,
+            size: ZoneDimens.appbarMainSize,
+          ),
+        ),
+      ),
+      body: Column(children: [const BuildTime(), WallControl(), Statistic()]),
+    );
+  }
```
Comment (juliomejia97) [inline lib/src/features/devices/presentation/training/training_screen.dart:40]: Good idea!

File: lib/src/features/devices/presentation/training/wall_control.dart [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+          IconButton(
+              onPressed: () {},
+              icon: const Icon(Icons.add_circle),
+              color: ZoneTheme.fontColor,
+              iconSize: ZoneDimens.appbarMainSize)
+        ],
+      ),
+    );
+  }
+}
```
Comment (juliomejia97) [inline lib/src/features/devices/presentation/training/wall_control.dart:35]: This has to be rendered according the number of walls selected. In the mid time create a variable and build this according the variable. With a .map or a .for you can perform this

**Review by juliomejia97 — CHANGES_REQUESTED** [review]
Awesome job! Really good job. Please review the changes. Also, run flutter `flutter analyze` in order to lint the code created

File: lib/src/features/devices/presentation/training/statistic/pass/overall_pass.dart (hunk 4 above)
Comment (BrayanRGarciaM) [reply lib/src/features/devices/presentation/training/statistic/pass/overall_pass.dart:26]: When we use a non const variable like widget.titleCategor, we can not put const

File: .flutter-plugins-dependencies [hunk 12]
```diff
@@ -1 +1 @@
...
\ No newline at end of file
+{"info":"This is a generated file; do not edit or check into version control.","plugins":{"ios":[{"name":"flutter_blue_plus","path":"C:\\\\Users\\\\ASUS\\\\Desktop\\\\LaZona\\\\flutter\\\\.pub-cache\\\\hosted\\\\pub.dartlang.org\\\\flutter_blue_plus-1.3.0\\\\","native_build":true,"dependencies":[]},{"name":"hexcolor","path":"C:\\\\Users\\\\ASUS\\\\Desktop\\\\LaZona\\\\flutter\\\\.pub-cache\\\\hosted\\\\pub.dartlang.org\\\\hexcolor-2.0.7\\\\","native_build":true,"dependencies":[]}],"android":[{"name":"flutter_blue_plus","path":"C:\\\\Users\\\\ASUS\\\\Desktop\\\\LaZona\\\\flutter\\\\.pub-cache\\\\hosted\\\\pub.dartlang.org\\\\flutter_blue_plus-1.3.0\\\\","native_build":true,"dependencies":[]},{"name":"hexcolor","path":"C:\\\\Users\\\\ASUS\\\\Desktop\\\\LaZona\\\\flutter\\\\.pub-cache\\\\hosted\\\\pub.dartlang.org\\\\hexcolor-2.0.7\\\\","native_build":true,"dependencies":[]}],"macos":[],"linux":[],"windows":[],"web":[]},"dependencyGraph":[{"name":"flutter_blue_plus","dependencies":[]},{"name":"hexcolor","dependencies":[]}],"date_created":"2022-08-17 21:42:30.004850","version":"3.0.2"}
```
Comment (juliomejia97) [inline .flutter-plugins-dependencies:1]: Remove this file from commits history

File: lib/src/features/devices/presentation/training/statistic/precision/precision.dart [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+  State<StatefulWidget> createState() {
+    return PrecisionStatisticState();
+  }
+}
+
+class PrecisionStatisticState extends State<PrecisionStatistic> {
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      margin: const EdgeInsets.all(20),
```
Comment (juliomejia97) [inline lib/src/features/devices/presentation/training/statistic/precision/precision.dart:23]: Review this value, maybe reduce to 10

File: lib/src/features/devices/presentation/training/training_screen.dart [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+        ),
+        flexibleSpace: const SafeArea(
+          child: Icon(
+            Icons.mode_standby_rounded,
+            color: whiteColor,
+            size: appbarMainSize,
+          ),
+        ),
+      ),
+      body: Column(children: [const BuildTime(), const WallSetting(), Statistic()]),
```
Comment (juliomejia97) [inline lib/src/features/devices/presentation/training/training_screen.dart:39]: Format

**Review by juliomejia97 — CHANGES_REQUESTED** [review]
Again, please before push run `flutter analyze` to make sure the checks are gonna pass

File: lib/src/core/strings.dart [hunk 15]
```diff
@@ -0,0 +1,12 @@
+const String MANUAL_CONTROL = "Manual Control";
```
Comment (juliomejia97) [inline lib/src/core/strings.dart:1]: // ignore_for_file: constant_identifier_names insert this line before

**Review by juliomejia97 — APPROVED** [review]

**Review by leonardomn12 — APPROVED** [review]

**Pull request merged**